avm/ptn/dev-ops/cicd-agents-and-runners/main.bicep

```
@@ -0,0 +1,1025 @@
+            name: 'AZP_URL'
+            secretRef: 'organization-url'
+          }
+          {
+            name: 'AZP_TOKEN'
+            secretRef: 'personal-access-token'
+          }
+        ]
+      }
+    ]
+    environmentResourceId: appEnvironment.outputs.resourceId
+    workloadProfileName: 'consumption'
+  }
+}
+module deploymentScriptPrivateDNSZone 'br/public:avm/res/network/private-dns-zone:0.5.0' = if (contains(
+  computeTypes,
+  'azure-container-app'
+) && selfHostedConfig.selfHostedType == 'azuredevops' && privateNetworking && empty(networkingConfiguration.computeNetworking.?deploymentScriptPrivateDnsZoneResourceId ?? '')) {
+  name: 'stgdsdnszone${namingPrefix}${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'privatelink.file.${environment().suffixes.storage}'
+    virtualNetworkLinks: [
+      {
+        virtualNetworkResourceId: networkingConfiguration.networkType == 'useExisting'
+          ? networkingConfiguration.virtualNetworkResourceId
+          : newVnet.outputs.resourceId
+      }
+    ]
+  }
+}
+
+module deploymentScriptStg 'br/public:avm/res/storage/storage-account:0.13.0' = if (contains(
+  computeTypes,
+  'azure-container-app'
+) && selfHostedConfig.selfHostedType == 'azuredevops' && privateNetworking) {
+  name: 'placeholderAgentDeploymentScript-${uniqueString(resourceGroup().id)}'
+  params: {
+    name: 'stgds${uniqueString(resourceGroup().id, selfHostedConfig.?placeHolderAgentName ?? 'placeHolderAgent',location,selfHostedConfig.agentsPoolName)}'
+    location: location
+    allowBlobPublicAccess: false
+    allowSharedKeyAccess: true
+    publicNetworkAccess: 'Disabled'
+    //Assigning Storage Blob Data Contributor role to the user assigned identity
+    roleAssignments: [
+      {
+        principalId: userAssignedIdentity.outputs.principalId
+        roleDefinitionIdOrName: '69566ab7-960f-475b-8e7c-b3118f30c6bd'
+        principalType: 'ServicePrincipal'
+      }
+    ]
+    privateEndpoints: [
+      {
+        service: 'file'
+        subnetResourceId: networkingConfiguration.networkType == 'useExisting'
+          ? '${networkingConfiguration.virtualNetworkResourceId}/subnets/${networkingConfiguration.computeNetworking.containerAppDeploymentScriptSubnetName}'
+          : filter(
+              newVnet.outputs.subnetResourceIds,
+              subnetId =>
+                contains(
+                  subnetId,
+                  networkingConfiguration.?containerAppDeploymentScriptSubnetName ?? 'app-deployment-script-subnet'
+                )
+            )[0]
+        privateDnsZoneGroupName: 'stgPrivateDNSZoneGroup'
+        privateDnsZoneResourceIds: !empty(networkingConfiguration.computeNetworking.?deploymentScriptPrivateDnsZoneResourceId ?? '')
+          ? [
+              networkingConfiguration.computeNetworking.?deploymentScriptPrivateDnsZoneResourceId ?? ''
+            ]
+          : [
+              deploymentScriptPrivateDNSZone.outputs.resourceId
+            ]
```

Like the other comment...
```suggestion
        privateDnsZoneResourceIds: [
          networkingConfiguration.computeNetworking.?deploymentScriptPrivateDnsZoneResourceId ?? deploymentScriptPrivateDNSZone.outputs.resourceId
        ]
```

---

Resolved